client/jetpack-connect/controller.js

```diff
@@ -165,35 +169,42 @@ export function offerResetRedirects( context, next ) {
 		return page.redirect( CALYPSO_PLANS_PAGE + selectedSite.slug );
 	}
 
-	// If the site already has a paid plan, skip the plans/products page and redirect back to wp-admin.
-	if ( hasPaidPlan ) {
-		debug(
-			'controller: offerResetRedirects -> redirecting to back to wp-admin because site already has a paid plan',
-			context.params
-		);
+	// If current user is not an admin (can't purchase plans), redirect the user to /posts if
+	// the connection was started within Calypso, otherwise redirect the user to wp-admin
+	if ( ! canPurchasePlans ) {
```

Comment:
The diff that GH generated for this file appears much more confusing than it really is.  All I did here is switch the order of these code blocks ( Switched the order of: `if ( ! canPurchasePlans ){` and `if ( hasPaidPlan ) {` code blocks ).  Then also I just added `return` statements.

Reply:
Yes, the GH diff is confusing. Took me a while to figure out the changes being made.

Reply:
Thanks for the review @manzoorwanijk!! 🙌 😄